test_common/harness/kernelHelpers.cpp

```diff
@@ -1511,22 +1511,33 @@ size_t get_min_alignment(cl_context context)
     return align_size;
 }
 
-cl_device_fp_config get_default_rounding_mode(cl_device_id device)
+cl_device_fp_config get_default_rounding_mode(cl_device_id device,
+                                              const cl_uint &param)
 {
+    if (param == CL_DEVICE_DOUBLE_FP_CONFIG)
+        test_error_ret(
+            -1,
+            "FAILURE: CL_DEVICE_DOUBLE_FP_CONFIG not supported by this routine",
+            0);
+
     char profileStr[128] = "";
     cl_device_fp_config single = 0;
-    int error = clGetDeviceInfo(device, CL_DEVICE_SINGLE_FP_CONFIG,
-                                sizeof(single), &single, NULL);
+    int error = clGetDeviceInfo(device, param, sizeof(single), &single, NULL);
```

Comment:
I'm getting here an error here on a device which doesn't support cl_khr_fp16, but trying to query `CL_DEVICE_HALF_FP_CONFIG`.

From a call stack starting from `generateRef` on test_printf.cpp:1194 then calling `get_default_rounding_mode(device, CL_DEVICE_HALF_FP_CONFIG)`